### DISCUSSION TITLE
Capturing agent logs during integration tests

### DISCUSSION THREAD

**Discussion by jlashner** [simonsobs/socs#723] [Q&A] [answered]
I think this is something I may have figured out before but can't seem to recreate it... Is there a standard way of capturing agent logs during integration tests to debug failures?

**Comment by BrianJKoopman** [comment, marked as answer]
Running locally it's possible to direct the agent logs to a file, then view that file. In the tests add `args=['--log-dir', './logs/']` to the agent runner fixture creation. For example:
```
run_agent = create_agent_runner_fixture(
    '../socs/agents/agent/agent.py', 'agent_name', args=['--log-dir', './logs/'])
```
